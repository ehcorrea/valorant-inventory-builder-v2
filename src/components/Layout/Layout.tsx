import { ImageBackground, ViewProps } from 'react-native';

import image from '@/assets/images/background.png';

export type LayoutProps = {
  children: React.ReactNode;
} & Pick<Partial<ViewProps>, 'onLayout'>;

export function Layout({ children, ...props }: LayoutProps) {
  return (
    <ImageBackground style={{ flex: 1 }} source={image} {...props}>
      {children}
    </ImageBackground>
  );
}
