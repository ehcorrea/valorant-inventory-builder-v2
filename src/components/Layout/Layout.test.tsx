import { View } from 'react-native';

import { render } from '@/test/utils';

import { Layout } from './Layout';

const setup = () =>
  render(
    <Layout>
      <View testID="children-view" />
    </Layout>
  );

describe('<Layout/>', () => {
  test('should render children', () => {
    const container = setup();
    expect(container.getByTestId('children-view')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
