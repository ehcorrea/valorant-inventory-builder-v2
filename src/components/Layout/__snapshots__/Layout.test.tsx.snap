// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> should render children 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    testID="children-view"
  />
</View>
`;
